### PR TITLE
Fallback on messenger to init if no storage permissions found

### PR DESCRIPTION
**Pull request by adrenak**
- On Android 30 and above, we check for Manage All Files permission before trying to read external json files. If permissions aren't available, we try cached/internal json files. If they don't exist, we skip json based init and wait for the messenger to receive the runtime settings summary and device status.

### CODE REVIEW

File: Assets/MXR.SDK/Runtime/Android/MXRAndroidSystem.cs [hunk 1]
```diff
@@ -439,7 +448,16 @@ public class MXRAndroidSystem : IMXRSystem {
...
-        void InitializeDeviceStatus() {
+        void TryInitializeDeviceStatus() {
+            // On Android30 and above, we check if we have storage access. Otherwise file read attempt will fail.
+            if (MXRAndroidUtils.AndroidSDKAsInt >= 30 && !MXRAndroidUtils.IsExternalStorageManager) {
+                if (LoggingEnabled)
+                    Debug.unityLogger.LogWarning(TAG, "Not initializing DeviceStatus using local json, "
+                    + "the SDK will fallback to trying to initialize it using the Android Messenger. This is not an error. "
+                    + EXTERNAL_STORAGE_MANAGER_WARNING_MSG);
+                return;
+            }
```
Comment (johnmanagexr) [inline Assets/MXR.SDK/Runtime/Android/MXRAndroidSystem.cs:459]: Can we make this a function and use it in both places this is called?

**Review by johnmanagexr — CHANGES_REQUESTED** [review]
Nice, this is a good start!

Either in this PR or a separate one, can we also handling writing deviceStatus info to a public location?

File: Assets/MXR.SDK/Runtime/Android/MXRAndroidSystem.cs (hunk 1 above)
Comment (Aleyum) [reply Assets/MXR.SDK/Runtime/Android/MXRAndroidSystem.cs:459]: + 1

**Comment by adrenak** [comment]
> Either in this PR or a separate one, can we also handling writing deviceStatus info to a public location?

Let's do that in the same PR. That'll allow startup to speed up too as we'll have files that we can read regardless of permissions.

File: Assets/MXR.SDK/Runtime/Android/MXRAndroidSystem.cs [hunk 2]
```diff
@@ -479,6 +558,25 @@ public class MXRAndroidSystem : IMXRSystem {
...
+            get {
+                // If we're on level 29 and below, we don't need external storage manager permissions
+                if (!MXRAndroidUtils.NeedsManageAllFilesPermission)
+                    return true;
+                else {
+                    if (MXRAndroidUtils.IsExternalStorageManager)
+                        return true;
+                    else
+                        return false;
+                }
```
Comment (johnmanagexr) [inline Assets/MXR.SDK/Runtime/Android/MXRAndroidSystem.cs:572]: nit: return MXRAndroidUtils.IsExternalStorageManager

**Review by johnmanagexr — CHANGES_REQUESTED** [review]
Looks great! Just a nit. Lets make sure we have a solid QA plan for this one outlined in the ticket?

File: Assets/MXR.SDK/Runtime/Android/MXRAndroidSystem.cs (hunk 2 above)
Comment (adrenak) [reply Assets/MXR.SDK/Runtime/Android/MXRAndroidSystem.cs]: Done!

**Comment by adrenak** [comment]
Going to write some QA notes on Clickup

**Review by Aleyum — APPROVED** [review]
LGTM, do we need to make any changes to the CD pipeline once this goes in ?

**Review by johnmanagexr — APPROVED** [review]

**Pull request merged**